spec/features/ncr/work_orders/post_approval_modification_spec.rb

```diff
@@ -0,0 +1,81 @@
+describe "post-approval modification" do
+  include ProposalSpecHelper
+
+  around(:each) do |example|
+    with_feature('DISABLE_SANDBOX_WARNING') do
+      example.run
+    end
+  end
+
+  let(:work_order) { create(:ncr_work_order) }
+
+  before do
+    work_order.setup_approvals_and_observers
+    fully_approve(work_order.proposal)
+
+    login_as(work_order.requester)
+  end
+
+  it "doesn't require re-approval for the amount being decreased" do
+    visit "/ncr/work_orders/#{work_order.id}/edit"
+    fill_in 'Amount', with: work_order.amount - 1
+    click_on 'Update'
+
+    work_order.reload
+    expect(work_order.status).to eq('approved')
+  end
+
+  it "can do end-to-end re-approval" do
+    visit "/ncr/work_orders/#{work_order.id}/edit"
+    fill_in 'Amount', with: work_order.amount + 1
+    click_on 'Update'
+
+    expect_budget_approvals_restarted
+
+    login_as(work_order.budget_approvers.first)
+    visit "/proposals/#{work_order.proposal.id}"
+    click_on 'Approve'
+
+    work_order.reload
+    expect(work_order.status).to eq('pending')
+    expect(work_order.proposal.root_approval.status).to eq('actionable')
+    expect(approval_statuses).to eq(%w(
+      approved
+      approved
+      actionable
+    ))
+
+    login_as(work_order.budget_approvers.second)
+    visit "/proposals/#{work_order.proposal.id}"
+    click_on 'Approve'
+
+    work_order.reload
+    expect(work_order.status).to eq('approved')
+    expect(work_order.proposal.root_approval.status).to eq('approved')
+    expect(approval_statuses).to eq(%w(
+      approved
+      approved
+      approved
+    ))
+  end
```

Comment:
@jessieay I hear you, though it would be very difficult to do exclusively through th's tricky to do exclusively through the UI because

* The code to verify this in the markup would be tricky, trying to distinguish which item words like "pending" and "approve" refer to. I could imagine this being very brittle.
    * This probably speaks to the 508 compliance of the Proposal show page, but that's a whole separate issue.
* It's checking for internal state (the `root_step`), which isn't reflected in the UI.
    * Checking for this directly revealed a bug, so I would be very reluctant to take it out.

Is there a compromise, like moving the checks into methods to describe what it's doing?